src/Domain/CoordinateMaps/Frustum.hpp

```diff
@@ -21,27 +21,238 @@ class er;
+ * \\
+ * \Delta^y_{\eta} &= \frac{1}{4}(y^{+\zeta}_{+\xi,+\eta} -
+ * y^{+\zeta}_{-\xi,-\eta} + y^{-\zeta}_{+\xi,+\eta} - y^{-\zeta}_{-\xi,-\eta})
+ * \\
+ * \Delta^y_{\eta\zeta} &= \frac{1}{4}(y^{+\zeta}_{+\xi,+\eta} -
+ * y^{+\zeta}_{-\xi,-\eta} - y^{-\zeta}_{+\xi,+\eta} + y^{-\zeta}_{-\xi,-\eta})
+ * \\
+ * \Sigma^z &= \frac{z_1 + z_2}{2}\\
+ * \Delta^z &= \frac{z_2 - z_1}{2}
+ * \f}
+ *
+ * The full map is then given by:
+ * \f[\vec{x}(\xi,\eta,\zeta) = \begin{bmatrix}
+ * \Sigma^x + \Delta^x_{\xi}\Xi + (\Delta^x_{\zeta} +
+ * \Delta^x_{\xi\zeta}\Xi)\zeta\\
+ * \Sigma^y + \Delta^y_{\eta}\mathrm{H} + (\Delta^y_{\zeta} +
+ * \Delta^y_{\eta\zeta}\mathrm{H})\zeta\\
+ * \Sigma^z + \Delta^z\zeta\\
+ * \end{bmatrix}\f]
+ *
+ * With Jacobian: \f[J =
+ * \begin{bmatrix}
+ * (\Delta^x_{\xi} + \Delta^x_{\xi\zeta}\zeta)\Xi' &
+ * 0 & \Delta^x_{\zeta}+ \Delta^x_{\xi\zeta}\Xi \\
+ * 0 & (\Delta^y_{\eta} + \Delta^y_{\eta\zeta}\zeta)\mathrm{H}' &
+ * \Delta^y_{\zeta} + \Delta^y_{\eta\zeta}\mathrm{H}\\
+ * 0 & 0 & \Delta^z \\
+ * \end{bmatrix}
+ * \f]
+ *
+ * ### Changing the distribution of gridpoints
+ * By default, \f$\zeta\f$ is used to linearly interpolate between the two
+ * bases, resulting in a uniform grid spacing between the bases of the frustum.
+ * This causes elements near the smaller base to the longer in the direction
+ * normal to the base, and elements near the larger base to be shorter in the
+ * direction normal to the base. We desire elements that have roughly equal
+ * sizes along each of their dimensions, and we obtain this through a
+ * projective map that redistributes the gridpoints in the \f$\zeta\f$
+ * direction.
+ *
+ * \image html BilinearVProjective.png "Comparison of mappings. (Noah Veltman)"
+ *
+ * As seen Veltman's [Warp-Off]
+ * (https://bl.ocks.org/veltman/8f5a157276b1dc18ce2fba1bc06dfb48),
+ * we can redistribute the gridpoints using a projective map.
+ * We can also see that in this particular example both maps can leave the
+ * elements distorted at one end.
+ * The projective map used in Frustum gets around this difficulty by using
+ * the parameter \f$w_{\delta} = \f$ `projective_scale_factor` to control
+ * the degree of projection, with \f$w_{\delta} = 1\f$ indicating no projection
+ * (bilinear).
+ *
+ * \image html ProjectionOntoPlane.png "Controlling the degree of projection."
+ *
+ * As seen above, \f$w_{\delta}\f$ controls the degree of projection of the
+ * \f$z_2\f$-base of the frustum (seen in purple) into a fourth spatial
+ * dimension \f$w\f$. The resulting grid (seen in red) is obtained by projecting
+ * back into the 3D plane at \f$w=1\f$. By default, `projective_scale_factor` is
+ * set to unity and the mapping from the logical cube to the frustum is an
+ * affine one. The user can choose to change the `projective_scale_factor`,
+ * increasing the foreshortening in the grid.
+ *
+ * The gridpoint distribution is determined by using \f$\zeta\f$ to linearly
```

Comment:
I do want to go into more detail explaining x tilde, but I feel that would go beyond the scope of the Frustum map (especially since this projection technique can be used in maps other than Frustum), and I don't know where I would add such documentation. Possibly an additional to the developer guide :smile: